src/main/java/seedu/address/model/product/Product.java

```diff
@@ -0,0 +1,120 @@
+package seedu.address.model.product;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import seedu.address.model.person.Name;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents Product to be sold by Suppliers and stored in Warehouse.
+ * Guarantees: immutable;
+ */
+public class Product {
+
+    private final Name productName;
+    private final int productQuantity;
+    private final Set<Tag> productTags = new HashSet<>();
+
+    /**
+     * Constructs a Product with a productName and the productQuantity associated with the product.
+     * Product objects constructed will not have any product tags associated with it.
+     *
+     * @param productName
+     * @param productQuantity
+     */
+    public Product(Name productName, int productQuantity) {
+        this.productName = productName;
+        this.productQuantity = productQuantity;
+    }
+
+    /**
+     * Constructs a Product with a productName and the tags (if any) associated with the products.
+     * Tags can be 0 or more than 1.
+     *
+     * @param productName
+     * @param productTags
+     */
+    public Product(Name productName, Set<Tag> productTags) {
+        this.productName = productName;
+        this.productQuantity = 0;
+        this.productTags.addAll(productTags);
+    }
+
+    /**
+     * Constructs a Product with a productName, the product's productQuantity
+     * and tags associated with the current product.
+     *
+     * @param productName
+     * @param productQuantity
+     * @param productTags
+     */
+    public Product(Name productName, int productQuantity, Set<Tag> productTags) {
+        this.productName = productName;
+        this.productQuantity = productQuantity;
+        this.productTags.addAll(productTags);
+    }
+
+    public Name getProductName() {
+        return this.productName;
+    }
+
+    public int getProductQuantity() {
+        return this.productQuantity;
+    }
+
+    public Set<Tag> getProductTags() {
+        return Collections.unmodifiableSet(this.productTags);
+    }
+
+    /**
+     * Creates a new Product with the change in quantity reflected.
+     *
+     * @param updatedQuantity
+     * @return
```

Comment:
Just a minor issue but perhaps this `@return` should be removed as well!

Reply:
Accepted, will remove `@return` and change "Creates" in first line of JavaDocs to "Returns"